Core/include/Acts/Fitter/KalmanFitter.hpp

```diff
@@ -463,12 +463,12 @@ class KalmanFitter {
           state.stepping.navDir = backward;
           state.stepping.stepSize = ConstrainedStep(state.options.maxStepSize);
           state.stepping.pathAccumulated = 0.;
+          state.stepping.jacToGlobal = BoundToFreeMatrix::Zero();
           // Reinitialize the stepping jacobian
           st.referenceSurface().initJacobianToGlobal(
-              state.options.geoContext, state.stepping.jacToGlobal,
+              state.options.geoContext, *state.stepping.jacToGlobal,
               state.stepping.pos, state.stepping.dir,
               st.filteredParameters(state.options.geoContext).parameters());
-          state.stepping.jacobian = BoundMatrix::Identity();
```

Comment:
Why is `jacobian` reinitialization removed here? I expect that the reinitialization is done for each material surface, so here as well.

Reply:
Ah, I see. Now, it's always `jacobian = jacFull;`.

Reply:
It's a modification of the workflow to avoid the issue that the jacobian represents the covariance transport from A to B while the covariance matrix is already transported to B. Instead the jacobian is always assigned to the last calculated one.